Documentation/CRDs/Cluster/external-cluster/topology-for-external-mode.md

```diff
@@ -0,0 +1,201 @@
+pool 11 'cd' replicated size 1 min_size 1 crush_rule 6 object_hash rjenkins pg_num 32 pgp_num 32 autoscale_mode on last_change 335 lfor 0/0/256 flags hashpspool,selfmanaged_snaps stripe_width 0 application rbd read_balance_score 1.19
+pool 12 'ef' replicated size 1 min_size 1 crush_rule 7 object_hash rjenkins pg_num 32 pgp_num 32 autoscale_mode on last_change 336 lfor 0/0/256 flags hashpspool,selfmanaged_snaps stripe_width 0 application rbd read_balance_score 1.06
+```
+
+!!! note
+   Require to run `ceph config set global  mon_allow_pool_size_one true` for creating replica-1 pools
+
+Review the specific crush rules:
+crush_rule 5
+```
+ {
+            "rule_id": 5,
+            "rule_name": "rule_host-osd0-hdd",
+            "type": 1,
+            "steps": [
+                {
+                    "op": "take",
+                    "item": -10,
+                    "item_name": "osd-0~hdd"
+                },
+                {
+                    "op": "choose_firstn",
+                    "num": 0,
+                    "type": "osd"
+                },
+                {
+                    "op": "emit"
+                }
+            ]
+}
+```
+
+Once the pools are ready with specific crush rules.
+
+Decide what will be the flags value based on the crush rules and kubernetes cluster nodes.
+
+--topology-pools -> Would be the 3 replica-1 pools, i.e ab,cd,ef
+--topology-failure-domain-label -> kubernetes cluster nodes label depending on the need.
+--topology-failure-domain-values -> kubernetes cluster nodes
+
+Current machine:
+```
+$ kubectl get nodes --show-labels
+NAME           STATUS   ROLES           AGE    VERSION   LABELS
+minikube       Ready    control-plane   3d2h   v1.27.4   beta.kubernetes.io/arch=amd64,beta.kubernetes.io/os=linux,kubernetes.io/arch=amd64,kubernetes.io/hostname=minikube,kubernetes.io/os=linux,minikube.k8s.io/commit=fd7ecd9c4599bef9f04c0986c4a0187f98a4396e,minikube.k8s.io/name=minikube,minikube.k8s.io/primary=true,minikube.k8s.io/updated_at=2024_03_04T16_18_42_0700,minikube.k8s.io/version=v1.31.2,node-role.kubernetes.io/control-plane=,node.kubernetes.io/exclude-from-external-load-balancers=
+minikube-m02   Ready    <none>          28h    v1.27.4   beta.kubernetes.io/arch=amd64,beta.kubernetes.io/os=linux,kubernetes.io/arch=amd64,kubernetes.io/hostname=minikube-m02,kubernetes.io/os=linux
+minikube-m03   Ready    <none>          28h    v1.27.4   beta.kubernetes.io/arch=amd64,beta.kubernetes.io/os=linux,kubernetes.io/arch=amd64,kubernetes.io/hostname=minikube-m03,kubernetes.io/os=linux
+```
+
+Then run the python script:
+```
+ python3 a.py --rbd-data-pool-name nitin --topology-pools ab,cd,ef --topology-failure-domain-label hostname --topology-failure-domain-values minikube,minikube-m02,minikube-m03
```

Comment:
a.py? Where is that script?

Reply:
updated, i used short from for testing thanks fro catching